runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowExecutionContext.java

```diff
@@ -245,6 +248,16 @@ public static class DataflowExecutionStateTracker extends ExecutionStateTracker
     private final ContextActivationObserverRegistry contextActivationObserverRegistry;
     private final String workItemId;
 
+    /**
+     * Metadata on the message whose processing is currently being managed by this tracker. If no
+     * message is actively being processed, activeMessageMetadata will be null.
+     */
+    private ActiveMessageMetadata activeMessageMetadata = null;
+
+    private MillisProvider clock = System::currentTimeMillis;
```

Comment:
make member variables final if not intended to be reassigned

Reply:
d6ef04f81684b1f0f8eb3e0f328de88c8a3f8f49